lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -290,28 +321,37 @@ def missing_key_helper(keys):
                 else:
                     keys[key] = "foobar"
 
-            missing_key_helper(keys)
+            if self.api_method == API_METHOD.POST:
+                missing_key_helper(keys)
```

Comment:
This should be done at the top level of the method, not in an inner loop, because the answer won't change!

Reply:
It's not the best time of day for me to be making this assertion, but I'm _pretty_ sure that the answer **will** change, because...isn't the value of `keys` different on every call?  (Also, this call is _after_ the innermost loop....)

Reply:
No, the point was that `self.api_method == API_METHOD.POST` won't change!

And, yeah, I wasn't entirely clear. I mean, if we're not going to actually deal with checking for missing keys in the URI or QUERY schemas, then we should skip this test when `self.api_method != API_METHOD.POST` up at the beginning rather than going through this whole thing doing nothing...

Reply:
Ah, right:  we could move the `if` test to just before the `for` statement at line 314 so that the outer loop is skipped if the method isn't `POST` (and then we could remove the `if` test from inside the loop).

@siddardh-ra, if you do that, then you should pull the `if` statement at line 329 into the same block as the `for` statement at line 314 (and remove the method test from it).